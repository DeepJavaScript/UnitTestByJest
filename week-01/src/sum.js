// function sum(a, b) {
//   return a + b;
// }

// ES6
const sum = (a, b) => a + b;

export default sum;