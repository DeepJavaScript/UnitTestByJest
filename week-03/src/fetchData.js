function fetchData(callback, type) {
  console.log(`--- fetchData: ${type} Start ---`);
  setTimeout(() => {
    callback('hi');
  }, 3000);
  console.log(`--- fetchData: ${type} End ---`);
}

function promiseFetchData(option) {
  return new Promise((resolve, reject) => {
    console.log(`--- Promise: Start ---`);
    setTimeout(() => {
      if (option.flag === 'success') resolve('hi');
      if (option.flag === 'fail') reject('error');
    }, 3000);
    console.log(`--- Promise: End ---`);
  });
}

export { fetchData, promiseFetchData };