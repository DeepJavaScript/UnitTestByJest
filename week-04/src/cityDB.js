let cityDB = [];

function initCityDB() {
  cityDB = ['Taipei', 'Tainan'];
}

function initCityDBPromise() {
  return new Promise((resolve, reject) => {
    console.log(`--- Promise: init DB Start ---`);
    setTimeout(() => {
      cityDB = ['Taipei', 'Tainan'];
      resolve();
    }, 3000);
    console.log(`--- Promise: init DB End ---`);
  });
}

function initCityDBCallback(callback) {
  console.log(`--- callback: init DB Start ---`);
  setTimeout(() => {
    cityDB = ['Taipei', 'Tainan'];
    callback();
  }, 3000);
  console.log(`--- callback: init DB End ---`);
}

function clearCityDB() {
  cityDB = [];
}

function clearCityDBPromise() {
  return new Promise((resolve, reject) => {
    console.log(`--- Promise: clear DB Start ---`);
    setTimeout(() => {
      cityDB = [];
      resolve();
    }, 3000);
    console.log(`--- Promise: clear DB End ---`);
  });
}

function clearCityDBCallback(callback) {
  console.log(`--- callback: clear DB Start ---`);
  setTimeout(() => {
    cityDB = [];
    callback();
  }, 3000);
  console.log(`--- callback: clear DB End ---`);
}

function isCity(city) {
  return cityDB.includes(city);
}

function addCity(city) {
  cityDB.push(city);
}

export {
  initCityDB,
  initCityDBPromise,
  initCityDBCallback,
  clearCityDB,
  clearCityDBPromise,
  clearCityDBCallback,
  isCity,
  addCity
}