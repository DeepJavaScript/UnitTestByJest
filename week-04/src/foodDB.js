let foodDB = [];

function initFoodDB() {
  foodDB = [
    { city: 'Taipei', name: 'Apple' },
    { city: 'Tainan', name: 'Banana' }
  ];
}

function initFoodDBPromise() {
  return new Promise((resolve, reject) => {
    console.log(`--- Promise: init DB Start ---`);
    setTimeout(() => {
      foodDB = [
        { city: 'Taipei', name: 'Apple' },
        { city: 'Tainan', name: 'Banana' }
      ];
      resolve();
    }, 3000);
    console.log(`--- Promise: init DB End ---`);
  });
}

function clearFoodDB() {
  foodDB = [];
}

function clearFoodDBPromise() {
  return new Promise((resolve, reject) => {
    console.log(`--- Promise: clear DB Start ---`);
    setTimeout(() => {
      foodDB = [];
      resolve();
    }, 3000);
    console.log(`--- Promise: clear DB End ---`);
  });
}

function isValidCityFoodPair(foodCity, foodName) {
  return foodDB.some(food => (food.city === foodCity) && (food.name === foodName));
}

export {
  initFoodDB,
  initFoodDBPromise,
  clearFoodDB,
  clearFoodDBPromise,
  isValidCityFoodPair
}