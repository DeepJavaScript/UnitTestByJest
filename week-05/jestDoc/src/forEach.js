function forEach(items, callback) {
  for (const item of items) {
    callback(item);
  }
}

export default forEach;