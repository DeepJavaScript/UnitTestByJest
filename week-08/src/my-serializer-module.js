module.exports = {
  serialize(val, config, indentation, depth, refs, printer) {
    return 'Pretty foo: ' + printer(val.foo);
  },
  test(val) {
    return val && val.hasOwnProperty('foo');
  }
};