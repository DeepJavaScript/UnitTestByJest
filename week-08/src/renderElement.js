function rendererLinkElement(url, linkName) {
  return `<a href="${url}">${linkName}</a>`;
}

function rendererImageElement(url, alt) {
  return `<img src="${url}" alt="${alt}">`;
}

function rendererMediaElement(link, image) {
  return `
    <section class="media">
      ${rendererImageElement(image.url, image.alt)}
      ${rendererLinkElement(link.url, link.name)}
    </section>`;
}

export {
  rendererLinkElement,
  rendererImageElement,
  rendererMediaElement
};
