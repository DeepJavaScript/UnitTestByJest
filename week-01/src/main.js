// const sum = require('./sum');

// ES6
import sum from './sum';

console.log(sum(1, 2));