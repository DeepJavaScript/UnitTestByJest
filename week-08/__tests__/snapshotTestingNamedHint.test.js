import { rendererLinkElement } from "../src/renderElement";

describe("render component", () => {
  it("render link element", () => {
    let url = 'https://titangene.github.io/';
    let linkName = 'Titangene Blog';
  
    let actual = rendererLinkElement(url, linkName);

    // expect(actual).toMatchSnapshot();
    // expect(actual).toMatchSnapshot();
    expect(actual).toMatchSnapshot('rendor Titangene Blog (First Test)');
    expect(actual).toMatchSnapshot('rendor Titangene Blog (Second Test)');
  });

  it("render link element", () => {
    let url = 'https://www.google.com/';
    let linkName = 'Google';
  
    let actual = rendererLinkElement(url, linkName);

    // expect(actual).toMatchSnapshot();
    // expect(actual).toMatchSnapshot();
    expect(actual).toMatchSnapshot('rendor Google (First Test)');
    expect(actual).toMatchSnapshot('rendor Google (Second Test)');
  });
});
