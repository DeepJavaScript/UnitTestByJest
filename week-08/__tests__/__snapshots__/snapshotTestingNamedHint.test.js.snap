// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component render link element: rendor Google (First Test) 1`] = `"<a href=\\"https://www.google.com/\\">Google</a>"`;

exports[`render component render link element: rendor Google (Second Test) 1`] = `"<a href=\\"https://www.google.com/\\">Google</a>"`;

exports[`render component render link element: rendor Titangene Blog (First Test) 1`] = `"<a href=\\"https://titangene.github.io/\\">Titangene Blog</a>"`;

exports[`render component render link element: rendor Titangene Blog (Second Test) 1`] = `"<a href=\\"https://titangene.github.io/\\">Titangene Blog</a>"`;
